src/Core/Form/IdentifiableObject/OptionProvider/ProductFormOptionsProvider.php

```diff
@@ -46,6 +46,7 @@ public function getOptions(int $id, array $data): array
             'product_type' => $data['header']['type'] ?? ProductType::TYPE_STANDARD,
             'virtual_product_file_id' => $data['stock']['virtual_product_file']['virtual_product_file_id'] ?? null,
             'active' => $data['header']['active'] ?? false,
+            'tax_rules_group_id' => $data['pricing']['retail_price']['tax_rules_group_id'] ?? null,
```

Comment:
Just to be sure did you check a use case where you change the tax rules and then submit the form with invalid values somewhere else? Just to make sure that on submit the help message is correctly adapted with the new value and not the initial one?

Reply:
Don't really get it what to check here. By invalid values somewhere you mean invalid values in product form?

Reply:
Yes, the idea is to force an invalid constraint in the form, for example use `<` in the reference input, this will be considered invalid, and the form is submitted and displayed again with inline errors. All the submitted values (including the reference) should be populated inside the form.

At this moment the previously selected tax rules should still be the one selected in the select input, and its help message should match the one you selected (country, potential state, tax rate, ...)

Reply:
yeah there was issue when some inputs are invalid but now its fixed.